tests/unit/confluence/test_pages.py

```diff
@@ -386,6 +395,69 @@ def test_delete_page_error(self, pages_mixin):
         with pytest.raises(Exception, match="Failed to delete page"):
             pages_mixin.delete_page(page_id)
 
+    def test_attach_content_success(self, pages_mixin):
+        """Test successfully attach content."""
+        # Arrange
+        page_id = "987654321"
+        content = b"Content to attach"
+        name = "test.pdf"
+
+        mock_response = MagicMock(spec=requests.Response)
+        mock_response.status_code = 200
+        pages_mixin.confluence.attach_content.return_value = mock_response
+
+        result = pages_mixin.attach_content(content=content, name=name, page_id=page_id)
+
+        # Assert
+        pages_mixin.confluence.attach_content.assert_called_once_with(
+            content=content, name=name, page_id=page_id
+        )
+
+        assert isinstance(result, ConfluencePage)
+        assert result.id == page_id
+
+    def test_attach_content_api_error(self, pages_mixin):
+        """Test error handling when attaching content."""
+        # Arrange
+        page_id = "987654321"
+        content = b"Content to attach"
+        name = "test.pdf"
+        exception_message = "Attachments are disabled or the calling user does not have permission to attach content."
+        pages_mixin.confluence.attach_content.side_effect = ApiError(exception_message)
+
+        # Act/Assert
+        result = pages_mixin.attach_content(content=content, name=name, page_id=page_id)
+        assert result is None
+
+    def test_attach_content_network_error(self, pages_mixin):
+        """Test error handling when attaching content due to network error."""
+        # Arrange
+        page_id = "987654321"
+        content = b"Content to attach"
+        name = "test.pdf"
+        exception_message = "Network error"
+        pages_mixin.confluence.attach_content.side_effect = RequestException(
+            exception_message
+        )
+
+        # Act/Assert
+        result = pages_mixin.attach_content(content=content, name=name, page_id=page_id)
+        assert result is None
+
+    def test_attach_content_unexpected_error(self, pages_mixin):
```

Comment:
Consider renaming this test to `test_attach_content_unhandled_exception_propagates` for slightly better clarity.